packages/core/src/connectors/gnosis.ts

```diff
@@ -19,19 +19,26 @@ export class GnosisConnector extends Connector<
 > {
   readonly id = 'gnosis'
   readonly name = 'Gnosis'
-  readonly ready = !__IS_SERVER__
+  ready = !__IS_SERVER__
 
   #provider?: SafeAppProvider
   #sdk?: SafeAppsSDK
   #safe?: SafeInfo
 
   constructor(config: { chains?: Chain[]; options?: SafeOpts }) {
     super({ ...config, options: config?.options })
-    if (!__IS_SERVER__) {
-      this.#sdk = new SafeAppsSDK(config.options)
-      // Auto connect on safe environment
-      this.connect()
-    }
+    this.#isSafeApp()
+      .then((isSafeApp) => {
+        if (!__IS_SERVER__ && isSafeApp) {
+          this.ready = true
+          this.#sdk = new SafeAppsSDK(config.options)
+          // Auto connect on safe environment
```

Comment:
Do you think it would be okay if I just add a check here within options for the provided flag `autoConnect` and if the user has flagged for it then execute else just return the sdk?

Reply:
or perhaps `doNotAutoConnect` would be a better naming convention and don't run `connect` when the flag is provided

Reply:
I think `doNotAutoConnect` is better

Reply:
😉 add `doNotAutoConnect`, plz check

Reply:
Looks great :)